plugin/src/main/java/hudson/plugins/swarm/DownloadClientAction.java

```diff
@@ -0,0 +1,30 @@
+package hudson.plugins.swarm;
+
+import hudson.Extension;
+import hudson.model.UnprotectedRootAction;
+import jenkins.model.Jenkins;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+
+@Extension
+public class DownloadClientAction implements UnprotectedRootAction {
+    public String getIconFileName() {
+        return null;
+    }
+
+    public String getDisplayName() {
+        return null;
+    }
+
+    public String getUrlName() {
+        return "swarm";
+    }
+
+    // serve static resources
+    public void doDynamic(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {

```

Comment:
Please add `@Restricted(NoExternalUse.class)` to indicate it is not a public API